src/ClientRuntime/Microsoft.Rest.ClientRuntime/ServiceClient.cs

```diff
@@ -212,6 +215,19 @@ protected ServiceClient()
         /// <summary>
         /// Initializes a new instance of the ServiceClient class.
         /// </summary>
+        /// <param name="httpClient">HttpClient</param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "Microsoft.Reliability",
+            "CA2000:Dispose objects before losing scope",
+            Justification = "The created objects should be disposed on caller's side")]
+        protected ServiceClient(HttpClient httpClient)
+        {
+            InitializeHttpClient(httpClient, CreateRootHandler());
```

Comment:
@markcowl with the overload now provided, we will not be setting any handlers. We do not want to set any handlers for the provided client.
The entire design is, all bets are off, when an httpClient is provided. We will not set any handlers from the provided httpClient. So whatever handlers that needs to be added should be added  before calling the new constructor.
@markcowl please can you provide an example of a scenario where we will not be able to set headers?